src/cogent3/core/new_sequence.py

```diff
@@ -1699,7 +1711,7 @@ def complement(self):
         will return list of keys.
         """
         return self.__class__(
-            moltype=self.moltype, seq=self.moltype.complement(self), info=self.info
+            moltype=self.moltype, seq=self.moltype.complement(str(self)), info=self.info
```

Comment:
everywhere you are using `str(self)` add a comment about checking using either `array(self)`.